sherpa/astro/data.py

```diff
@@ -94,66 +95,81 @@ class DataOgipResponse(Data1DInt):
-            checked here, since there are issues involving numerical effects
-            (e.g. when checking that two bins do not overlap), as well as
-            uncertainty over what possible  behavior is seen in released
-            data products for missions. The current set of checks are:
-
-              - ehi > elo for each bin
-              - elo is monotonic (ascending or descending)
-              - when emin is set, the lowest value in elo is >= 0,
-                otherwise it is > 0.
-              - ethresh (if set) is less than the minimum value in ENERG_HI
-
-            A failed check raises a DataErr exception.
+        Several checks are made, to make sure the parameters follow
+        the OGIP standard. At present a failed check can result in
+        either a warning message being logged, or an error raised.
+        It was felt that raising an error in all cases would not be
+        helpful to a user, who can't (easily) change the response
+        files.
 
-            """
+        Parameters
+        ----------
+        label : str
+            The response file identifier.
+        elo, ehi : numpy arrays
+            The input ENERG_LO and ENERG_HI arrays. They are assumed
+            to be one-dimensional and have the same number of elements.
+        ethresh : None or float, optional
+            If None, then elo must be greater than 0. When set, the
+            start bin can have a low-energy edge of 0; it is replaced
+            by ethresh. If set, ethresh must be greater than 0.
+            An error is raised if ethresh is larger than the upper-edge
+            of the first bin (only if the lower edge has been replaced).
+
+        Returns
+        -------
+        elo, ehi : numpy arrays
+            The validated energy limits. These can be the input arrays
+            or a copy of them. At present the ehi array is the same as
+            the input array, but this may change in the future.
+
+        Notes
+        -----
+        Only some of the constraints provided by the OGIP standard are
+        checked here, since there are issues involving numerical effects
+        (e.g. when checking that two bins do not overlap), as well as
+        uncertainty over what possible  behavior is seen in released
+        data products for missions. The current set of checks are:
+
+          - ehi > elo for each bin
+          - elo is monotonic (ascending or descending)
+          - when emin is set, the lowest value in elo is >= 0,
+            otherwise it is > 0.
+          - ethresh (if set) is less than the minimum value in ENERG_HI
+
+        """
 
         rtype = self._ui_name
 
         if ethresh is not None and ethresh <= 0.0:
             raise ValueError("ethresh is None or > 0")
 
+        # Only display the first warning, since it doesn't really
+        # help users to have multiple warnings for the same response.
+        #
+        warned = False
```

Comment:
I believe this can be done in a cleaner way using the warnings framework, but I'll have to check.